Client/Frontend/Settings/AppSettingsOptions.swift

```diff
@@ -107,14 +120,82 @@ class SyncNowSetting: WithAccountSetting {
     }
 
+    private lazy var troubleshootButton: UIButton = {
+        let troubleshootButton = UIButton(type: UIButtonType.RoundedRect)
+        troubleshootButton.setTitle(Strings.FirefoxSyncTroubleshootTitle, forState: .Normal)
+        troubleshootButton.addTarget(self, action: #selector(self.troubleshoot), forControlEvents: .TouchUpInside)
+        troubleshootButton.tintColor = UIConstants.TableViewRowActionAccessoryColor
+        troubleshootButton.titleLabel?.font = DynamicFontHelper.defaultHelper.DefaultSmallFont
+        troubleshootButton.sizeToFit()
+        return troubleshootButton
+    }()
+
+    private lazy var warningIcon: UIImageView = {
+        let imageView = UIImageView(image: UIImage(named: "AmberCaution"))
+        imageView.sizeToFit()
+        return imageView
+    }()
+
+    private lazy var errorIcon: UIImageView = {
+        let imageView = UIImageView(image: UIImage(named: "RedCaution"))
+        imageView.sizeToFit()
+        return imageView
+    }()
+
+    // TODO: This needs to be changed to the actual URL when we have it
+    private let syncSUMOURL = NSURL(string: "https://support.mozilla.org")!
+
+    @objc private func troubleshoot() {
+        let viewController = SettingsContentViewController()
+        viewController.url = syncSUMOURL
+        settings.navigationController?.pushViewController(viewController, animated: true)
+    }
+
     override func onConfigureCell(cell: UITableViewCell) {
         cell.textLabel?.attributedText = title
-        cell.detailTextLabel?.attributedText = status
+        cell.textLabel?.numberOfLines = 0
+        cell.textLabel?.lineBreakMode = .ByWordWrapping
+        if let syncStatus = profile.syncManager.syncState {
+            switch syncStatus {
+            case .Bad(_):
+                // add the red warning symbol
+                // add a link to the MANA page
+                cell.detailTextLabel?.attributedText = nil
+                cell.accessoryView = troubleshootButton
+                addIcon(errorIcon, toCell: cell)
+            case .Stale(_):
+                // add the amber warning symbol 
+                // add a link to the MANA page
+                cell.detailTextLabel?.attributedText = nil
+                cell.accessoryView = troubleshootButton
+                addIcon(warningIcon, toCell: cell)
+            case .Good:
+                cell.detailTextLabel?.attributedText = status
+                fallthrough
+            default:
+                cell.accessoryView = nil
+            }
+        } else {
+            cell.accessoryView = nil
+        }
         cell.accessoryType = accessoryType
-        cell.accessoryView = nil
         cell.userInteractionEnabled = !profile.syncManager.isSyncing
     }
 
+    private func addIcon(image: UIImageView, toCell cell: UITableViewCell) {
+        cell.addSubview(image)
+
+        cell.textLabel?.snp_updateConstraints { make in
+            make.leading.equalTo(image.snp_trailing).offset(10)
```

Comment:
When text wraps, this constraint is ignored. To test, force syncStatus to be `SyncState.Stale(message: String(format:Strings.FirefoxSyncPartialTitle, Strings.localizedStringForSyncComponent("bookmarks") ?? ""))`. Not sure why this is happening yet